python_modules/dagster/dagster/config/post_process.py

```diff
@@ -186,7 +188,7 @@ def _recurse_in_to_array(context: TraversalContext, config_value: Any) -> Evalua
     if not config_value:
         return EvaluateValueResult.for_value([])
 
-    if context.config_type.inner_type.kind != ConfigTypeKind.NONEABLE:
+    if cast(ConfigType, context.config_type.inner_type).kind != ConfigTypeKind.NONEABLE:  # type: ignore
```

Comment:
cast and type: ignore? would expect one or the other

Reply:
I had done this bc there are multiple type issues on that line (the `cast` covers only one of them) but you're right, better to have everything covered by `type: ignore`.